packages/bundler-plugin-core/src/index.ts

```diff
@@ -86,16 +92,25 @@ export function sentryUnpluginFactory({
     });
 
     function handleRecoverableError(unknownError: unknown) {
-      pluginExecutionTransaction.setStatus("internal_error");
-
-      if (options.errorHandler) {
-        if (unknownError instanceof Error) {
-          options.errorHandler(unknownError);
+      sentrySession.status = "abnormal";
+      try {
+        if (options.errorHandler) {
+          try {
+            if (unknownError instanceof Error) {
+              options.errorHandler(unknownError);
+            } else {
+              options.errorHandler(new Error("An unknown error occured"));
+            }
+          } catch (e) {
+            sentrySession.status = "crashed";
+            throw e;
+          }
         } else {
-          options.errorHandler(new Error("An unknown error occured"));
+          sentrySession.status = "crashed";
+          throw unknownError;
         }
-      } else {
-        throw unknownError;
+      } finally {
+        sentryHub.endSession();
```

Comment:
> Do we need this in addition to the process.on endSession call?

Yes. beforeExit is not called on exceptions and the like: https://nodejs.org/api/process.html#event-beforeexit

> Do we need this in addition to the process.on endSession call?

No because that variable is only there to prevent an infinite loop with beforeExit which will be invoked again when the event loop is empty.

I'll add comments! :)

Reply:
thx for explaining :)